File: February/CPP/Find the town Judge/Find the town judge.cpp
class Solution {
public:
    int findJudge(int n, vector<vector<int>>& trust) {
       vector<int> in(n+1),out(n+1);
       for(auto x: trust){
           ++in[x[1]];
           ++out[x[0]];
       }
       for(int i=1;i<=n;i++){
           if(in[i]==n-1 && out[i]==0) return i;
       }
       return -1;
    }
};

File: February/Python/Find the Town Judge/Find the Town Judge.py
class Solution:
    def findJudge(self, n: int, trust: List[List[int]]) -> int:
        incoming = defaultdict(int)
        outgoing = defaultdict(int)

        for src, dst in trust:
            outgoing[src] += 1
            incoming[dst] += 1

        for i in range(1, n + 1):
            if outgoing[i] == 0 and incoming[i] ==n - 1:
                return i
        return -1
